application/lib/view/screens/ScreenEditorScreen.dart

```diff
@@ -0,0 +1,245 @@
+import 'dart:async';
+
+import 'package:application/blocs/ScreenBloc.dart';
+import 'package:application/di.dart';
+import 'package:application/model/EditorScreenElement.dart';
+import 'package:application/model/Screen.dart';
+import 'package:application/routes.dart';
+import 'package:application/view/screens/BaseScreen.dart';
+import 'package:application/view/widgets/NotifyDIalog.dart';
+import 'package:application/view/widgets/RoundedTextField.dart';
+import 'package:flutter/material.dart';
+import 'package:font_awesome_flutter/font_awesome_flutter.dart';
+
+class ScreenEditorScreen extends BaseScreen {
+  ScreenEditorScreen(this.screen) {
+    for (var widget in screen.screenContent) {
+      screenContent.add(EditorScreenElement(
+          widgetType: widget['type'],
+          key: widget['key'],
+          position: int.parse(widget['position'])));
+    }
+  }
+
+  final ScreenBloc screenBloc = di.getDependency<ScreenBloc>();
+  final StreamController<EditorScreenElement> selectedElement =
+      StreamController<EditorScreenElement>.broadcast();
+
+  final Screen screen;
+
+  final List<EditorScreenElement> screenContent = [];
+
+  final List<String> widgets = ['Text', 'Flat button', 'Label'];
```

Comment:
This should get injected and not coded in here, we need some kind of pattern that can decouple this.